qcodes/instrument_drivers/signal_hound/USB_SA124B.py

```diff
@@ -1,166 +1,327 @@
         self.add_parameter('rbw',
                            label='Resolution Bandwidth',
                            unit='Hz',
                            initial_value=1e3,
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Numbers())
+                           vals=vals.Numbers(),
+                           parameter_class=TraceParameter)
         # vbw Video bandwidth in Hz. VBW must be less than or equal to RBW.
-        #  VBW can be arbitrary. For best performance use RBW as the VBW.
+        # VBW can be arbitrary. For best performance use RBW as the VBW.
+        # what does this even do though? nataliejpg
         self.add_parameter('vbw',
                            label='Video Bandwidth',
                            unit='Hz',
                            initial_value=1e3,
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Numbers())
+                           vals=vals.Numbers(),
+                           parameter_class=TraceParameter)
+
+        self.add_parameter('reject_image',
+                           label='Reject image',
+                           unit='',
+                           initial_value=True,
+                           parameter_class=TraceParameter,
+                           get_cmd=None,
+                           docstring="Apply software filter to remove "
+                                     "undersampling mirroring",
+                           vals=vals.Bool())
 
         self.openDevice()
         self.device_type()
 
-        t1 = time()
-        # poor-man's connect_message. We could overwrite get_idn
-        # instead and use connect_message.
-        print('Initialized SignalHound in %.2fs' % (t1-t0))
+        self._prepare_measurement()
+        sweep_len, start_freq, stepsize = self.QuerySweep()
+        self.add_parameter(name='trace',
+                           sweep_len=sweep_len,
+                           start_freq=start_freq,
+                           stepsize=stepsize,
+                           parameter_class=FrequencySweep)
+        self.npts._save_val(sweep_len)
+
+        self.connect_message()
+
+    def _update_trace(self) -> None:
+        """
+        Private method to sync changes of the
+        frequency axis to the setpoints of the
+        trace parameter.
+        """
+        sweep_info = self.QuerySweep()
+        self.npts._save_val(sweep_info[0])
+        self.trace.set_sweep(*sweep_info)
+
+    def _sync_parameters(self) -> None:
+        """
+        Sync parameters consists of five parts
+            1. Center span configuration (freqs and span)
+            2. Acquisition configuration
+                lin-scale/log-scale
+                avg/max power
+            3. Configuring the external 10MHz reference
+            4. Configuration of the mode that is being used
+            5. Configuration of the tracking generator (not implemented)
```

Comment:
I agree that is a good idea to refactor this out but that should imho go together with grouping the parameters so that you can change the parameters that need to be changed in one go and then sync them together